maddr/maddr.go

```diff
@@ -26,28 +26,32 @@ func FromDIDService(svc didcore.Service) ([]MoneyAddress, error) {
 	}
 
 	maddrs := make([]MoneyAddress, len(svc.ServiceEndpoint))
-
 	for i, se := range svc.ServiceEndpoint {
-		urn, err := urn.Parse(se)
+		maddr, err := FromURN(svc.ID, se)
 		if err != nil {
 			return nil, fmt.Errorf("invalid money address: %w", err)
 		}
+		maddrs[i] = *maddr
+	}
+	return maddrs, nil
+}
 
-		delimIDX := strings.IndexRune(urn.NSS, ':')
-		if delimIDX == -1 {
-			return nil, fmt.Errorf("invalid money address. expected urn:[currency]:[protocol]:[pss]. got %s", se)
-		}
-
-		maddr := MoneyAddress{
-			URN:      urn,
-			ID:       svc.ID,
-			Currency: urn.NID,
-			Protocol: urn.NSS[:delimIDX],
-			PSS:      urn.NSS[delimIDX+1:],
-		}
+func FromURN(serviceID string, maddr string) (*MoneyAddress, error) {
+	urn, err := urn.Parse(maddr)
+	if err != nil {
+		return nil, fmt.Errorf("invalid money address: %w", err)
+	}
 
-		maddrs[i] = maddr
+	delimIDX := strings.IndexRune(urn.NSS, ':')
+	if delimIDX == -1 {
+		return nil, fmt.Errorf("invalid money address. expected urn:[currency]:[protocol]:[pss]. got %s", maddr)
 	}
 
-	return maddrs, nil
+	return &MoneyAddress{
+		URN:      urn,
+		ID:       serviceID,
```

Comment:
we should create an issue to remove `ID`

Reply:
Just drop it entirely?